src/execution_plan/ops/op_merge.c

```diff
@@ -105,7 +105,7 @@ OpBase *NewMergeOp(const ExecutionPlan *plan, EntityUpdateEvalCtx *on_match,
 			op->on_match[i].record_idx = OpBase_Modifies((OpBase *)op, op->on_match[i].alias);
 		}
 	}
-
+	OpBase_RegisterAsWriter((OpBase *) op);
```

Comment:
Not necessary if `op->on_match == NULL`.

Reply:
changed